src/redux/productsSlice.js

```diff
@@ -0,0 +1,113 @@
+import { createSlice, createAsyncThunk } from "@reduxjs/toolkit"
+import firebaseClient from "firebase/client"
+
+const initialState = {
+  data: [],
+  products: [],
+  loadingStatus: "idle",
+  error: null,
+}
+
+const productsSlice = createSlice({
+  name: "products",
+  initialState,
+  reducers: {},
+})
+
+export const createNewProduct = createAsyncThunk(
+  "products/createNewProduct",
+  async (payload) => {
+    try {
+      await _createNewProduct(
+        payload.productName,
+        payload.productPrice,
+        payload.productPhoto,
+        payload.productRoom,
+        payload.productDuration
+      )
+      return payload
+    } catch (error) {
+      console.error("error", error)
+    }
+  }
+)
+async function _createNewProduct(
```

Comment:
I would have requested a product object instead of spread parameters.
Then handle the possible errors inside the function. Because its the concern of this function.

Reply:
Since it’s functional right now, let’s go ahead and push this as is for v1.